src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -1119,6 +1130,7 @@ def join(
             "right",
```

Comment:
also expand the check here https://github.com/snowflakedb/snowpark-python/pull/2095/files#diff-b123de92fc315a358428fa2a33390e78a0b8ee641420d7f0f67fe65dc58315daL1101, that the newly added left_macth_col and right_match_col also have to be part of the projected column